policy/lib/result_helper.rego

```diff
@@ -23,13 +23,59 @@ _basic_result(chain, failure_sprintf_params) := {
 }
 
 _code(chain) := code if {
-	pkg_path := chain[count(chain) - 1].path
-	pkg_name := pkg_path[count(pkg_path) - 1]
+	rule_path := chain[0].path
+	pkg_name := _pkg_name(rule_path)
+
+	# Todo someday maybe: Conftest supports denies named deny_some_name,
+	# so we could use that format and ditch the short name annotation.
 	rule_name := _rule_annotations(chain).custom.short_name
+
+	# Put them together
 	code := sprintf("%s.%s", [pkg_name, rule_name])
 }
 
 # The first entry in the chain always points to the active rule, even if it has
 # no declared annotations (in which case the annotations member is not present).
-# Thus, result_helper assumes every rule defines annotations.
+# Thus, result_helper assumes every rule defines annotations. At the very least
+# custom.short_name must be present.
 _rule_annotations(chain) := chain[0].annotations
+
+# This is meant to match the special handling done in ec-cli, see here:
+# https://github.com/enterprise-contract/ec-cli/blob/014a488a4/internal/opa/rule/rule.go#L161-L186
+_pkg_name(rule_path) := name if {
+	# Seems to not work if I keep assigning to a single var, so
+	# that's why the many different pN vars.
+
+	# Strip off the first element which is always "data"
+	p1 := _left_strip_elements(["data"], rule_path)
+
+	# Strip off policy.release or policy.pipeline to match what ec-cli
+	# does. (There are some edge cases where the behavior is not exactly
+	# the same, but I think this version is better.)
+	p2 := _left_strip_elements(["policy", "release"], p1)
```

Comment:
I think we should try to come up with a generic approach where we don't need to rely on supporting a specific set of namespaces. Could be done later. Could you file an issue for that?

Reply:
Filed https://issues.redhat.com/browse/EC-864 .

Reply:
Thanks!